pymeasure/instruments/anaheimautomation/stepmotorcontroller.py

```diff
@@ -0,0 +1,251 @@
+        """ Reset the position as counted by an externally connected encoder to 0.
+        """
+        self.write("ET")
+
+    def stop(self):
+        """Method that stops all motion on the motor controller."""
+        self.write(".")
+
+    def go(self):
+        """Method that sends the G command to the controller to tell the controller to turn the 
+           motor the number of steps previously set with the steps property."""
+
+        self.write("G")
+
+    def step(self, steps, direction):
+        """Similar to the go() method, but also sets the number of steps and the direction in the same method.
+        
+        :param steps: Number of steps to clock
+        :param direction: Value to set on the direction property before sending the go command to the controller.
+        """
+        self.steps = steps 
+        self.direction = direction
+        self.write("G")
+
+    def slew(self, direction):
+        """Sends the slew command to the motor controller. This tells the controller to clock the stepper motor until a stop command is sent or a limit switch is reached.
+        
+        :param direction: value to set on the direction property before sending the slew command to the controller.
+        """
+        self.direction = direction
+        self.write("S") 
+
+    async def async_step(self, steps, direction, **kwargs):
+        """ Asynchronous implementation of the step() method. This method can be awaited and will not return until the controller completes its step operation.
+        """
+        # call the normal step method #
+        self.step(steps, direction)
+        # wait until the step operation completes. # 
+        await self.wait_for_completion(**kwargs)
+
+    def write(self, command):
+        """Override the instrument base write method to add the motor controller's id to the command string.
+        
+        :param command: command string to be sent to the motor controller.
+        """
+        self.adapter.write("@{}".format(self.idn) + command)
+
+    def values(self, command, **kwargs):
+        """ Override the instrument base values method to add the motor controller's id to the command string.
+        
+        :param command: command string to be sent to the motor controller.
+        """
+        return self.adapter.values("@{}".format(self.idn) + command, **kwargs)
+        
+    def ask(self, command):
+        """ Override the instrument base ask method to add the motor controller's id to the command string.
+
+        :param command: command string to be sent to the instrument
+        """
+        return self.adapter.ask("@{}".format(self.idn) + command)
+   
+    async def wait_for_completion(self, interval=0.5, threshold=3):
+        """ Query the motor controller's step (or encoder) position and only return when inactivity is detected. Inactivity is 
+            defined as when the previous queried position and current queried position match for "threshold" number of times.
```

Comment:
Upon further inspection of the command set, the controller does in fact have a "busy" query which allows one to directly check if the controller is moving a motor. This query would be definitely be better suited in this case. I went ahead and removed this function since it is an async function, but added a `busy` property to perform this query anyway.